code/modules/mob/living/update_status.dm

```diff
@@ -49,7 +49,14 @@
 
 // Whether the mob is capable of talking
 /mob/living/can_speak()
-	return !(silent || (disabilities & MUTE) || is_muzzled())
+	if(!(silent || (disabilities & MUTE)))
+		if(is_muzzled())
+			var/obj/item/clothing/mask/muzzle/M = wear_mask
+			if(M.mute == 2)
```

Comment:
Make sure to update this flag check, and also to make it an & check instead of an == check.